assign-doi-datacite/src/main/java/no/unit/nva/doi/datacite/clients/DataCiteClient.java

```diff
@@ -56,37 +55,39 @@
     public static final String ERROR_SETTING_DOI_URL_TEMPLATE =
         ERROR_SETTING_DOI_URL
             + DOI_AND_HTTP_STATUS_TEMPLATE_ENTRIES;
+
+    private static final String HTTP_STATUS_LOG_MESSAGE = "{}";
     public static final String ERROR_CREATING_DOI_TEMPLATE =
         ERROR_CREATING_DOI
             + PREFIX_TEMPLATE_ENTRY
-            + HTTP_STATUS_LOG_TEMPLATE;
+            + HTTP_STATUS_LOG_TEMPLATE
+            + HTTP_STATUS_LOG_MESSAGE;
     private static final Logger logger = LoggerFactory.getLogger(DataCiteClient.class);
-    private final DataCiteMdsConnectionFactory mdsConnectionFactory;
+    private final DataCiteConnectionFactory dataCiteApiConnectionFactory;
     private final DataCiteConfigurationFactory configFactory;
 
     public DataCiteClient(DataCiteConfigurationFactory configFactory,
-                          DataCiteMdsConnectionFactory mdsConnectionFactory) {
+                          DataCiteConnectionFactory connectionFactory
+    ) {
         this.configFactory = configFactory;
-        this.mdsConnectionFactory = mdsConnectionFactory;
+        this.dataCiteApiConnectionFactory = connectionFactory;
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public Doi createDoi(URI customerId, String metadataDataCiteXml) throws ClientException {
-        var prefix = configFactory.getConfig(customerId).getCustomerDoiPrefix();
+    public Doi createDoi(URI customerId) throws ClientException {
+        DataCiteMdsClientSecretConfig customerConfigInfo = (DataCiteMdsClientSecretConfig)
+            configFactory.getConfig(customerId);
         try {
-            var response = prepareAuthenticatedDataCiteConnection(customerId)
-                .postMetadata(prefix, metadataDataCiteXml);
-            if (isUnsuccessfulResponse(response)) {
-                logger.error(ERROR_CREATING_DOI_TEMPLATE, prefix, response.statusCode());
-                throw new CreateDoiException(prefix, response.statusCode());
-            }
-            String createMetadataResponseBody = response.body();
-            var doi = extractDoiPrefixAndSuffix(createMetadataResponseBody);
-            return doi;
-        } catch (IOException | URISyntaxException | InterruptedException e) {
+            DataCiteRestConnection connection = prepareAuthenticatedDataCiteRestConnection(customerId,
+                customerConfigInfo);
```

Comment:
And we are not passing this customerConfigInfo like this here.

Reply:
Do you want add the  DataCiteMdsClientSecretConfig object in the constuctor in the DataCiteConnectionFactory?

Reply:
I am not sure I understand the point. Do you mean the `prepareAuthenticateDataCiteRestConnection` should not use customerConfigInfo or should it obtain it through the configFactory?

Reply:
When using a DataCiteRestConnection, you are not supposed to know that you have to pass a configuration, that's one of the reasons we have the factory for retrieve an authenticated DataCiteRestConnection for a customer.

The idea is that `DataCiteRestConnection` is already authenticated.